### DISCUSSION TITLE
I need a simple and nice looking 404 page, but it always shows the default 404 page

### DISCUSSION THREAD

**Discussion by BytesZero** [web-infra-dev/rspress#1328] [Ideas]
## 工程和构建
![image](https://github.com/user-attachments/assets/697cc482-6c5d-46b1-9aea-7531c5949fa0)

## 期望样式

- https://flutterads.top/404
![image](https://github.com/user-attachments/assets/ade1da95-b8bf-4738-9956-5a460611a67d)

## 实际效果
- https://flutterads.top/405
- https://flutterads.top/test/111

![image](https://github.com/user-attachments/assets/573dd723-b969-49b3-be70-61bbdfbf64b6)

## 我的想法

我可以编写任意 404.mdx ，然后默认的 404 就会是我的 404.mdx 内容，因为默认的不好看，我想在 404 中做更多的事情，感谢

**Comment by BytesZero** [comment]
我也尝试查看了文档和PR，最终没有看懂如何简单的构建一个漂亮的 404 页面
https://github.com/web-infra-dev/rspress/issues/1171
https://rspress.dev/zh/api/config/config-frontmatter#pagetype
https://rspress.dev/zh/plugin/official-plugins/client-redirects#%E6%B3%A8%E6%84%8F
https://rspress.dev/zh/guide/advanced/custom-theme#3-%E8%87%AA%E5%AE%9A%E4%B9%89%E7%BB%84%E4%BB%B6

**Comment by Timeless0911** [comment]
To customize a 404 page, you can directly add a `theme` folder in your root dir, and add a `index.tsx` file like this

```tsx
import Theme from "rspress/theme";

const NotFoundLayout = () => <div>404</div>;

export default {
  ...Theme,
  NotFoundLayout,
};

export * from "rspress/theme";
```

You can write any react component for `NotFoundLayout`.

**Reply by BytesZero** [reply]
@Timeless0911 
- 我看到过这个内容，但是他真的不简单
- 我认为他是复杂的，对于前端开发者来讲可能还好
- 对于只想写 md 专注于内容的人，还是有些麻烦
- 我的想法就是如果有 404.mdx 那么就替换到默认的 404 页面。（这样我只需创建一个 404.mdx 即可）

**Reply by Timeless0911** [reply]
这件事情没有单纯的替换这么简单呢，还涉及运行时的 fallback 以及 i18n、dark mode 等，最终的呈现形式叫做主题。这个设计和其他众多静态站点方案一样，都需要对主题进行重载。
- Vitepress https://vitepress.dev/guide/extending-default-theme#layout-slots
- Vuepress https://vuepress.github.io/advanced/theme.html
- Docusaurus https://docusaurus.io/docs/swizzling#swizzling-theme-components

**Reply by BytesZero** [reply]
了解啦，从简单的角度，我认为专注于内容编写的这样最为简单，我的理解是 404 状态的重定向就可以满足需求。
